src/components/Api.js

```diff
@@ -0,0 +1,95 @@
+export class Api {
+  constructor(options) {
+    this.options = options;
+    this.baseUrl = options.baseUrl;
+    this.headers = options.headers;
+  }
+
+  getCards() {
+    return fetch(`${this.baseUrl}/cards`, {
+      headers: this.headers,
+    });
```

Comment:
You need to parse the response inside `Api`. You shouldn't do that in the other files.  It’s a violation of **Single responsibility** principle

Reply:
According to the task, you need to check if the response is `ok` to invoke an error if the response is not `ok`.


```js
.then(res => {
                if (res.ok) {
                    return res.json();
                }
                return Promise.reject(`Error ${res.status}`);
            })

```

Reply:
```js
.then(res => {
                if (res.ok) {
                    return res.json();
                }
                return Promise.reject(`Error ${res.status}`);
            })

```

The code of `ok` checking will be repeated in **every** request of the project. To avoid it you need to make a special method `_checkResponse` that will check any response from the server if it’s ok or not.


```js
_checkResponse(res) {
        // here is the code of the checking
    }

```

And now you can replace redundant repeating with the only line:


```js
 .then(this._checkResponse)

```

Please, pay attention that you need to pass only the **reference** for the method rather than the call. 

**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.